manticore/core/smtlib/operators.py

```diff
@@ -124,7 +124,7 @@ def ULE(a, b):
 
 
 def EXTRACT(x, offset, size):
-    if isinstance(x, BitVec):
+    if isinstance(x, BitVec) and not isinstance(offset, BitVec):
```

Comment:
This is clearly better!
But now I'm thinking that if `offset` happen to be a BitVec then the else will fail to generate a BitVec of `size` bits.

Reply:
Hmm, fair point. Could call BitVecExtract on the expression with offset 0 and size size?

Reply:
Hmm I do not really know if the EXTRACT symbol actually supports symbolic size. 
Maybe Operators.EXTRACT needs to force/ request a concrete size. And any user of it must concretize the size before doing the EXTRACT. ... .

Reply:
Oh I don't think size can be symbolic, that will fail. But with the new change, offset should be able to.

Reply:
Right! Even from the evm.BYTE size if safely concrete.  nvm that.  
https://github.com/trailofbits/manticore/blob/402de84c0686b892c7ef987882f330cad565760e/manticore/platforms/evm.py#L1603-L1606

Reply:
Yea, that's the one that was failing with dynamic offset, but size is always 8.